static/js/tasks_browse.js

```diff
@@ -291,8 +291,14 @@ $(document).ready(function() {
         showAssignWorkerModal()
     });
 
-    $('#delete-tasks-modal .cancel-delete').on('click', function() {
+    // reset the local variable as well as the state
+    function retSelectedTask() {
         selectedTask = undefined;
+        taskBrowse.setSelectedTask(undefined);
```

Comment:
Also consider: `taskBrowse.setSelectedTask(selectedTask);`

Reply:
Will do, thank you.